src/test/rmc/Intrinsics/Assert/zero_valid_panic.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// rmc-verify-fail
+
+#![feature(core_intrinsics)]
+use std::intrinsics;
+
+// The code below attempts to zero-initialize type `&i32`, causing the intrinsic
+// `assert_zero_valid` to generate a panic during compilation.
+fn main() {
```

Comment:
Should we also have a test with an expected pass (e.g. zero initializing a `u8`)?

Reply:
The test in `src/test/rmc/Intrinsics/Assert/zero_valid.rs` (path updated in this PR) already does this for several types, including a user-defined struct.